### PR TITLE
Modify S6333: Improve the general text

### PR DESCRIPTION
**Pull request by loris-s-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [x] logical errors and incorrect information
- [x] information gaps and missing content
- [x] text style and tone
- [x] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S6333/ask-yourself.adoc [hunk 1]
```diff
@@ -1,8 +1,6 @@
 == Ask Yourself Whether
 
-The public API:
+* The underlying API uses an authentication method independantly from the API Gateway.
+* The underlying API is actually inaccessible from the Internet thanks to other mechanisms.
 
-* exposes sensitive data like personal information.
-* can be used to perform sensitive operations.
```
[the remark this answers is not in the document]
Comment (hendrik-buchwald-sonarsource) [reply rules/S6333/ask-yourself.adoc:6]: I agree. But I think the new version is not that easy to understand the way it is formulated. I don't have a concrete idea of how it should be done but I think it still needs another iteration. To be more concrete, I think the "There is a risk if you answered no to all of those questions" makes it more difficult to understand

Maybe you can try to use the new AI features that we have access to now to come up with an idea that still follows the "There is a risk if you answered yes to any of those questions" pattern?

File: rules/S6333/description.adoc [hunk 2]
```diff
@@ -1 +1,8 @@
-A public API, which can be requested by any authenticated or unauthenticated identities, can lead to unauthorized actions and information disclosures. 
+Creating APIs without authentication unnecessarily opens an attack vector on
+the target infrastructure.
+
+Unless another authentication method is used, attackers now have the
+opportunity to attempt attacks against the underlying API. +
+This means attacks both on the functionnality provided by the API and its
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/description.adoc:6]:
```suggestion
This means attacks both on the functionality provided by the API and its
```

File: rules/S6333/description.adoc [hunk 3]
```diff
@@ -1 +1,8 @@
-A public API, which can be requested by any authenticated or unauthenticated identities, can lead to unauthorized actions and information disclosures. 
+Creating APIs without authentication unnecessarily opens an attack vector on
+the target infrastructure.
+
+Unless another authentication method is used, attackers now have the
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/description.adoc:4]:
```suggestion
Unless another authentication method is used, attackers have the
```

File: rules/S6333/description.adoc [hunk 4]
```diff
@@ -1 +1,8 @@
-A public API, which can be requested by any authenticated or unauthenticated identities, can lead to unauthorized actions and information disclosures. 
+Creating APIs without authentication unnecessarily opens an attack vector on
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/description.adoc:1]: That works. Or maybe it increases the attack surface. Both is fine for me.

File: rules/S6333/recommended.adoc [hunk 5]
```diff
@@ -1,4 +1,21 @@
 == Recommended Secure Coding Practices
 
-It's recommended to restrict API access to authorized entities, unless the API offers a non-sensitive service designed to be public.
+In general, prefer limiting API access to a specific set of people or entities.
+
+AWS provides multiple methods to do so, whose use changes depending on the
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/recommended.adoc:5]: I am not 100% sure what is meant by "whose use changes".

File: rules/S6333/recommended.adoc [hunk 6]
```diff
@@ -1,4 +1,21 @@
...
+underlying need.
+
+Using "AWS_IAM" uses the IAM service usually used to manage access to AWS
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/recommended.adoc:8]: "Using ... uses ... usually used" does not sound too good.

File: rules/S6333/recommended.adoc [hunk 7]
```diff
@@ -1,4 +1,21 @@
...
+
+Using "COGNITO_USER_POOLS" uses AWS's Cognito directory services to manage user
+authentication for AWS-independent need, such as user authentication and
+authorization for web and mobile applications. +
+It is an OpenID Connect (OIDC) identity provider (IdP). 
+
+Using "CUSTOM" means using a Lambda authorizer, formerly known as a custom
+authorizer. It can be used to authorize an API using an external OIDC provider.
+
+More information about Api Gateway access control is available
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/recommended.adoc:19]:
```suggestion
More information about API Gateway access control is available
```

File: rules/S6333/recommended.adoc [hunk 8]
```diff
@@ -1,4 +1,21 @@
...
+https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-to-api.html[here].
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/recommended.adoc:20]: Maybe move this link to the resource documentation instead?

**Review by hendrik-buchwald-sonarsource — CHANGES_REQUESTED** [review]
See comments.

File: rules/S6333/description.adoc (hunk 4 above)
Comment (loris-s-sonarsource) [reply rules/S6333/description.adoc:1]: Hmm yeah attack surface might be more used than attack vectors, by dev people.

```suggestion
Creating APIs without authentication unnecessarily increases the attack surface on
```

File: rules/S6333/ask-yourself.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (loris-s-sonarsource) [reply rules/S6333/ask-yourself.adoc:6]: I changed it to one sentence.

File: rules/S6333/recommended.adoc [hunk 9]
```diff
@@ -2,20 +2,8 @@
...
-It is an OpenID Connect (OIDC) identity provider (IdP). 
-
-Using "CUSTOM" means using a Lambda authorizer, formerly known as a custom
-authorizer. It can be used to authorize an API using an external OIDC provider.
-
-More information about API Gateway access control is available
-https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-to-api.html[here].
+AWS provides multiple methods to do so:
 
+* `AWS_IAM`, to use Standard AWS IAM roles and policies.
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/recommended.adoc:7]: Is "standard" part of a name? After a quick Google search, I don't think so. So then I would write it in lower-case.

File: rules/S6333/recommended.adoc [hunk 10]
```diff
@@ -2,20 +2,8 @@
...
+* `COGNITO_USER_POOLS`, to use a customizable OpenID Connect (OIDC) identity providers (IdP).
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6333/recommended.adoc:8]: Mix of singular and plural (provider**s**).

**Review by hendrik-buchwald-sonarsource — APPROVED** [review]
LGTM, but some minor details are left.

File: rules/S6333/recommended.adoc (hunk 9 above)
Comment (loris-s-sonarsource) [reply rules/S6333/recommended.adoc:7]:
```suggestion
* `AWS_IAM`, to use standard AWS IAM roles and policies.
```

File: rules/S6333/recommended.adoc (hunk 10 above)
Comment (loris-s-sonarsource) [reply rules/S6333/recommended.adoc:8]:
```suggestion
* `COGNITO_USER_POOLS`, to use customizable OpenID Connect (OIDC) identity providers (IdP).
```

**Pull request merged**